### PR TITLE
[test] Add OpInfo for aten.native_group_norm

### PR DESCRIPTION
**Pull request by benjaminglass1**
Stack from [ghstack](https://github.com/ezyang/ghstack/tree/0.14.0) (oldest at bottom):
* #184791
* #182907
* #183980
* #185151
* #183946
* #183001
* __->__ #183000

This is a prerequisite to the refactoring of the decomp and CompositeExplicitAutograd for this op.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben @jataylo

### CODE REVIEW

**Comment by jansel** [comment]
@claude review these changes

**Review by cyyever — COMMENTED** [review]
Doing a pass since this has sat without review for a week and is blocking #183001. Test-only PR, but a few items before approval:

1. **`supports_expanded_weight=True` contradicts the xfails.** All three `TestExpandedWeightFunctional` tests are marked `expectedFailure` with the note that grad doesn't support expanded weights. Either flip the flag to `False`, or split the support flag so forward and grad can be reported separately.

2. **`assert_jit_shape_analysis=True` contradicts `TestJit::test_variant_consistency_jit` xfail.** Pick one.

3. **`check_gradient=False` for fp16 needs a tracking issue.** The comment "more investigation is needed" in `test/inductor/test_torchinductor_opinfo.py` (added twice, for cuda and xpu) will become permanent without an issue link. Convention is `# TODO(#NNNNN): ...`.

4. **`reference_group_norm` zero-size behavior changed subtly.** The refactored wrapper now returns `np.empty_like(input)` (uninitialized) for `Y` in the zero-numel path, where the old code returned a NaN-filled array via `inp.mean(axis=-1)`. Worth a quick grep of `reference_group_norm` callers to confirm nothing depends on the NaN.

**Comment by benjaminglass1** [comment]
@cyyever Thanks for your review! I've removed the misconfigured OpInfo parameters (I believe I based them on one of the other `native_*_norm` functions, as a starting point for this PR, and then never revisited them).

WRT the `check_gradient` on f16, I already know the issue being flagged there, and will have a PR open in this stack today to address it. I only did it that way so that the PR in the stack that fixed the _forward_ pass could remove an XFAIL for clarity. Would you still prefer a tracking issue?

The only place `reference_group_norm` is used is within this file, for `nn.functional.group_norm` OpInfo, which should have the same semantics as this function. The CI pipeline is still passing, so I think that should be safe.

**Review by cyyever — APPROVED** [review]

File: test/functorch/test_vmap.py [hunk 1]
```diff
@@ -4440,6 +4440,7 @@ def sample_vmap_out_dim_numpy_split_copy_with_int(
                         sample.kwargs["memory_format"] == torch.channels_last
                     ),
                 ),
+                xfail("native_group_norm"),
```
Comment (Skylion007) [inline test/functorch/test_vmap.py:4441]: We can definitely reuse the group_norm rules...

File: test/functorch/test_ops.py [hunk 2]
```diff
@@ -2055,6 +2062,7 @@ def reference(primals, cotangents, primals_tangents, cotangents_tangents):
                 # TODO: implement batching rule
                 xfail("_batch_norm_with_update"),
                 xfail("native_dropout_backward"),
+                xfail("native_group_norm"),
```
Comment (Skylion007) [inline test/functorch/test_ops.py:2039]: Same as group_norm batch rule. :)

**Review by Skylion007 — COMMENTED** [review]
Number of xfails can be massively reduced by having it reuse non-native group norm functorch registration.

**Comment by benjaminglass1** [comment]
@Skylion007 I'll put a PR in the stack that does your suggestions! This one is intended to be the bare-minimum "add the OpInfo and make the pipeline pass" PR.

File: test/functorch/test_vmap.py (hunk 1 above)
Comment (benjaminglass1) [reply test/functorch/test_vmap.py:4441]: @Skylion007 See https://github.com/pytorch/pytorch/pull/184791 ! I'll also work on the backward pass. Would you mind approving so I can merge?

**Comment by benjaminglass1** [comment]
CI failures (currently) appear to be unrelated to this PR and any missing test skips. Could a maintainer approve this? @Skylion007 @zou3519 @kshitij12345

**Comment by benjaminglass1** [comment]
Rebased on `main` to pickup ongoing work on MPS `native_group_norm` which would block merge anyway.

**Review by Skylion007 — APPROVED** [review]
The MPS consistency issue is real sadly.

**Comment by benjaminglass1** [comment]
@Skylion007 Yes, I anticipated that there might be additional failures, but didn't have time to chase them down before the weekend. I'm building on a Mac rn to run the tests and add an appropriate set of XFAILs, then I'll merge.
